### PR TITLE
[icons] feat: data-connection icon

### PR DESCRIPTION
**Pull request by jheadrick2**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Changes proposed in this pull request:

New Icon! 

#### Screenshot

image

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @jheadrick2! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: resources/icons/16px/bp3-data-connection-16.svg [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adidahiya) [inline resources/icons/16px/bp3-data-connection-16.svg]: the name of this file does not match the icon name. it must be `data-connection.svg`, as shown in the [build failure](https://app.circleci.com/pipelines/github/palantir/blueprint/376/workflows/7bcbd439-021d-4f93-b20e-aae13fd5ae47/jobs/35040):

```
lerna ERR! yarn run compile stderr:
(node:248) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/circleci/project/resources/icons/16px/data-connection.svg'
```

File: packages/icons/resources/icons/icons.json [hunk 2]
```diff
@@ -3490,5 +3490,11 @@
...
         "tags": "region, geo, fence",
         "group": "interface"
-    }
+    },
+    {
+        "displayName": "Data Connection",
+        "iconName": "data-connection",
+        "tags": "connectivity, connection, database, data, status, health",
+        "group": "interface"
+      }
```
Comment (adidahiya) [inline packages/icons/resources/icons/icons.json]:
```suggestion
    }
```

**Comment by adidahiya** [comment]
[docs preview](https://35070-71939872-gh.circle-artifacts.com/0/packages/docs-app/dist/index.html#icons)

the icon doesn't show up correctly, it renders a blank box:
 
image

File: resources/icons/16px/data-connection.svg [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>bp3-data-connection</title>
+    <defs>
```
Comment (adidahiya) [inline resources/icons/16px/data-connection.svg]: please re-export these icons with simpler SVG definitions similar to the other icons in the repo. see my comment here https://github.com/palantir/blueprint/pull/4132#discussion_r427577589

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: resources/icons/20px/data-connection.svg [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    </defs>
+    <g id="bp3-data-connection" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <mask id="mask-2" fill="white">
+            <use xlink:href="#path-1"></use>
+        </mask>
+        <use id="Mask" fill="#5C7080" xlink:href="#path-1"></use>
+    </g>
```
Comment (adidahiya) [inline resources/icons/20px/data-connection.svg]:
```suggestion
    <g id="bp3-data-connection" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M2,11.9 C2.93468865,12.5742236 4.3389668,13.1169578 6.02286947,13.4360955 C6.00757687,13.6215864 6,13.8099083 6,14 C6,16.5467063 7.359945,18.77577 9.39321594,20.00055 C5.40462513,19.9811517 2.16168117,18.9170543 2.00586153,17.5992952 L2,17.5 L2,11.9 Z M13,8 C16.315,8 19,10.685 19,14 C19,17.315 16.315,20 13,20 C9.685,20 7,17.315 7,14 C7,10.685 9.685,8 13,8 Z M14,9 L10,15 L12.5,15 L12,19 L16,13 L13.5,13 L14,9 Z M16.9995124,4.4 L16.9999712,8.2549186 C15.8661123,7.46390447 14.487116,7 12.9997725,7 C9.73921075,7 6.99933978,9.22941453 6.22127541,12.2470809 C3.81335585,11.8555478 2.12414114,11.0452198 2.00655427,10.1049899 L2,10 L2,4.4 C3.52495043,5.5 6.29986022,6.25 9.4997562,6.25 C12.6996522,6.25 15.474562,5.5 16.9995124,4.4 Z M9.4997562,0 C13.6371217,0 16.9995124,1.125 16.9995124,2.5 C16.9995124,3.875 13.6371217,5 9.4997562,5 C5.3498911,5 2,3.875 2,2.5 C2,1.125 5.3498911,0 9.4997562,0 Z" id="path-1"></path>
    </g>
```

**Comment by adidahiya** [comment]
[docs preview](https://35233-71939872-gh.circle-artifacts.com/0/packages/docs-app/dist/index.html#icons) looks good

**Review by adidahiya — APPROVED** [review]

**Pull request merged**